### PR TITLE
Extent SourceTask for check and format tasks.

### LINKED ISSUE #85: Make the Ktlint task a SourceTask
Instead of a `JavaExec` task, the `Ktlint` task should be [SourceTask](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceTask.html).

### PR DESCRIPTION
**Pull request by Tapchicoma**
Closes #85

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
Unfortunately, this is going to be an API breaking change, unless @bamboo,@wolfs, or @eskatos has any brilliant ideas as to how not to have this be an API breaking change.

**Comment by Tapchicoma** [comment]
If you are talking about this:
```kotlin
    @InputFiles
    @SkipWhenEmpty
    @PathSensitive(PathSensitivity.RELATIVE)
    override fun getSource(): FileTree { return super.getSource() }
```
I have to add it otherwise task becomes non-relocatable (not support caching).

Original `getSource()` method doesn't have any `@PathSensitive` annotation.

**Comment by JLLeitschuh** [comment]
@Tapchicoma Yea, I know. That's why I was asking for feedback.

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt [hunk 1]
```diff
@@ -97,11 +96,7 @@ open class KtlintCheckTask @Inject constructor(
     ): (JavaExecSpec) -> Unit = { javaExecSpec ->
         javaExecSpec.classpath = classpath
         javaExecSpec.main = "com.github.shyiko.ktlint.Main"
-        javaExecSpec.args(sourceDirectories.flatMap { dir ->
-            KOTLIN_EXTENSIONS.map { extension ->
-                "${dir.path}/**/*.$extension"
-            }
-        })
+        javaExecSpec.args(getSource())
```
[the remark this answers is not in the document]
Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt]: yes, this actually returns paths to kotlin files, not directories passed to task.

[the remark this answers is not in the document]
Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt]: Check added `init {}` block that adds include filters to this source task. I assume internally it filters out all passed sources.

**Pull request merged**